beginner_source/torchtext_custom_dataset_tutorial.py

```diff
@@ -0,0 +1,383 @@
+#
+# * At line 5, we set `special_first=True`. Which means `<pad>` will get index 0, `<sos>` index 1,
+#   `<eos>` index 2, and <unk> will get index 3 in the vocabulary.
+# * At line 7, we set default index as index of `<unk>`. That means if some word is not in
+#   vocbulary, we will use `<unk>` instead of that unknown word.
+#
+# Similarly, we will build vocabulary for target sentences:
+
+targetVocab = build_vocab_from_iterator(
+    get_tokens(dataPipe,1),
+    min_freq=2,
+    specials= ['<pad>', '<sos>', '<eos>', '<unk>'],
+    special_first=True
+)
+targetVocab.set_default_index(targetVocab['<unk>'])
+
+# %%
+# Note that the example above shows how can we add special tokens to our vocabulary. The
+# special tokens may change based on the requirements.
+#
+# Now, we can verify that special tokens are placed at the beginning and then other words.
+# In the below code, `sourceVocab.get_itos()` returns a list with tokens at index based on
+# vocabulary.
+
+print(sourceVocab.get_itos()[:9])
+
+# %%
+# Numericalize sentences using vocabulary
+# ---------------------------------------
+# After building the vocabulary, we need to convert our sentences to corresponding indices.
+# Let us define some functions for this:
+
+def get_transform(vocab):
+    """
+    Create transforms based on given vocabulary. The returned transform is applied to sequence
+    of tokens.
+    """
+    text_tranform = T.Sequential(
+        ## converts the sentences to indices based on given vocabulary
+        T.VocabTransform(vocab=vocab),
+        ## Add <sos> at beginning of each sentence. 1 because the index for <sos> in vocabulary is
+        # 1 as seen in previous section
+        T.AddToken(1, begin=True),
+        ## Add <eos> at beginning of each sentence. 2 because the index for <eos> in vocabulary is
+        # 2 as seen in previous section
+        T.AddToken(2, begin=False)
+    )
+    return text_tranform
+
+# %%
+# Now, let us see how to use the above function. The function returns an object of `Transforms`
+# which we will use on our sentence. Let us take a random sentence and check the working of
+# the transform:
+
+tempList = list(dataPipe)
+someSentence = tempList[798][0]
+print("Some sentence=", end="")
+print(someSentence)
+transformedSentence = get_transform(sourceVocab)(eng_tokenize(someSentence))
+print("Transformed sentence=", end="")
+print(transformedSentence)
+indexToString = sourceVocab.get_itos()
+for index in transformedSentence:
+    print(indexToString[index], end=" ")
+
+# %%
+# In the above code,:
+#
+#   * At line 2, we take a source sentence from list that we created from dataPipe at line 1
+#   * At line 5, we get a transform based on a source vocabulary and apply it to a tokenized
+#     sentence. Note that transforms take list of words and not a sentence.
+#   * At line 8, we get the mapping of index to string and then use it get the transformed
+#     sentence
```

Comment:
Remove 2 spaces in front of these bullets so the indentation is consistent with the rest of the tutorial.

Reply:
I have made changes as suggested.